types/address_test.go

```diff
@@ -58,6 +63,13 @@ func (s *addressTestSuite) testMarshal(original, res interface{}, marshal func()
 	s.Require().Equal(original, res)
 }
 
+func (s *addressTestSuite) testMarshalYAML(original, res interface{}, marshal func() (interface{}, error), unmarshal func([]byte) error) {
```

Comment:
I agree with you. I think that the test suite has its place in the unit tests as all, however, in this case, it is useless because it doesn't maintain any state.  I only used it for my tests because previous tests had already used the test suite and for consistency  I decided to use it too. Here, table tests would be more appropriate. I can remove the test suite and implement table tests in a separate PR because the file is almost 1000 rows. What do you think about that?